kube-controllers/pkg/controllers/node/hostendpoints.go

```diff
@@ -17,300 +17,425 @@ package node
-		config:    c,
-		client:    client,
-		nodeCache: make(map[string]*libapi.Node),
+const (
+	timer                        = 5 * time.Minute
+	hostEndpointNameSuffix       = "auto-hep"
+	defaultHostEndpointInterface = "*"
+)
+
+type hostEndpointTracker struct {
+	hostEndpointsByNode map[string]map[string]*api.HostEndpoint
+}
+
+func (t *hostEndpointTracker) addHostEndpointForNode(hostEndpoint *api.HostEndpoint) {
+	if t.hostEndpointsByNode[hostEndpoint.Spec.Node] == nil {
+		t.hostEndpointsByNode[hostEndpoint.Spec.Node] = make(map[string]*api.HostEndpoint)
+	}
+	t.hostEndpointsByNode[hostEndpoint.Spec.Node][hostEndpoint.Name] = hostEndpoint
+}
+
+func (t *hostEndpointTracker) deleteHostEndpointForNode(hostEndpointName string) {
+	var nodeName string
+	for _, heps := range t.hostEndpointsByNode {
+		for hepName, hep := range heps {
+			if hepName == hostEndpointName {
+				nodeName = hep.Spec.Node
+				delete(t.hostEndpointsByNode[hep.Spec.Node], hepName)
+			}
+		}
+	}
+
+	if len(t.hostEndpointsByNode[nodeName]) == 0 {
+		delete(t.hostEndpointsByNode, nodeName)
+	}
+}
+
+func (t *hostEndpointTracker) getHostEndpoint(hostEndpointName string) *api.HostEndpoint {
+	for _, heps := range t.hostEndpointsByNode {
+		for hepName, hep := range heps {
+			if hepName == hostEndpointName {
+				return hep
+			}
+		}
+	}
+
+	return nil
+}
+
+func (t *hostEndpointTracker) getHostEndpointsForNode(nodeName string) []*api.HostEndpoint {
+	var hostEndpoints []*api.HostEndpoint
+	for _, hep := range t.hostEndpointsByNode[nodeName] {
+		hostEndpoints = append(hostEndpoints, hep)
+	}
+
+	return hostEndpoints
+}
+
+func (t *hostEndpointTracker) getAllHostEndpoints() []*api.HostEndpoint {
+	var hostEndpoints []*api.HostEndpoint
+	for _, heps := range t.hostEndpointsByNode {
+		for _, hep := range heps {
+			hostEndpoints = append(hostEndpoints, hep)
+		}
+	}
+
+	return hostEndpoints
+}
+
+func NewAutoHEPController(cfg config.NodeControllerConfig, client client.Interface) *autoHostEndpointController {
+	return &autoHostEndpointController{
+		config:              cfg,
+		client:              client,
+		nodeCache:           make(map[string]*libapi.Node),
+		nodeUpdates:         make(chan string, 1),
+		syncerUpdates:       make(chan interface{}),
+		syncChan:            make(chan interface{}, 1),
+		hostEndpointTracker: hostEndpointTracker{hostEndpointsByNode: make(map[string]map[string]*api.HostEndpoint)},
 	}
-	return ctrl
 }
 
 type autoHostEndpointController struct {
-	rl         workqueue.TypedRateLimiter[any]
-	config     config.NodeControllerConfig
-	client     client.Interface
-	nodeCache  map[string]*libapi.Node
-	syncStatus bapi.SyncStatus
+	config              config.NodeControllerConfig
+	client              client.Interface
+	syncStatus          bapi.SyncStatus
+	nodeCache           map[string]*libapi.Node
+	nodeUpdates         chan string
+	syncerUpdates       chan interface{}
+	syncChan            chan interface{}
+	hostEndpointTracker hostEndpointTracker
```

Comment:
```suggestion
	autoHEPs hostEndpointTracker
```

reading the code, I think it helps if this is more clearly marked as only containing auto HEPs.